### PR TITLE
fix(server): workflow date filter, make end date inclusive

### PR DESCRIPTION
**Pull request by kigrup**
## Description

In the new Workflow Date filter element, the "End Date" parameters create a new date at hour/minute 0:00. This results in the "End Date" day being excluded from the results.

image

To make the filter include assets from the provided end date, we should change the comparison to "<" and make it against the following day. The Date constructor already handles overflow, so for example creating a Date with day 32 would not result in an error.

## How Has This Been Tested?

- [x] Created a workflow testing this exact use case and it correctly included the asset in the end date.

<details><summary><h2>Screenshots (if appropriate)</h2></summary>

N/A

</details>

## Checklist:

- [x] I have carefully read CONTRIBUTING.md
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation if applicable
- [x] I have no unrelated changes in the PR.
- [x] I have confirmed that any new dependencies are strictly necessary.
- [x] I have written tests for new code (if applicable)
- [x] I have followed naming conventions/patterns in the surrounding code
- [x] All code in `src/services/` uses repositories implementations for database calls, filesystem operations, etc.
- [x] All code in `src/repositories/` is pretty basic/simple and does not have any immich specific logic (that belongs in `src/services/`)

## Please describe to which degree, if any, an LLM was used in creating this pull request.

None

### CODE REVIEW

File: server/test/medium/specs/workflow/workflow-core-plugin.spec.ts [hunk 1]
```diff
@@ -428,6 +428,53 @@ describe('core plugin', () => {
...
   });
 
+  describe('assetDateFilter', () => {
+    it('should favorite assets created during the first 7 days of a specific year and month', async () => {
+      const { user } = await ctx.newUser();
+      const [{ asset: asset1 }, { asset: asset2 }, { asset: asset3 }, { asset: asset4 }] = await Promise.all([
+        ctx.newAsset({ ownerId: user.id, localDateTime: new Date(2000, 2, 31, 13, 0, 0, 0) }),
+        ctx.newAsset({ ownerId: user.id, localDateTime: new Date(2000, 3, 1, 2, 0, 0, 0) }),
+        ctx.newAsset({ ownerId: user.id, localDateTime: new Date(2000, 3, 7, 14, 0, 0, 0) }),
+        ctx.newAsset({ ownerId: user.id, localDateTime: new Date(2000, 3, 8, 0, 0, 0, 0) }),
```
Comment (danieldietzler) [inline server/test/medium/specs/workflow/workflow-core-plugin.spec.ts]: I would only do `new Date('2000-04-01')` and `new Date('2000-04-07T23:59:59Z')`

**Comment by arturict** [comment]
Looks clean. For date filter behavior, a short test case matrix would help reviewers verify edge conditions quickly.

File: server/test/medium/specs/workflow/workflow-core-plugin.spec.ts (hunk 1 above)
Comment (kigrup) [reply server/test/medium/specs/workflow/workflow-core-plugin.spec.ts]: Done

**Review by danieldietzler — APPROVED** [review]
Thank you!
